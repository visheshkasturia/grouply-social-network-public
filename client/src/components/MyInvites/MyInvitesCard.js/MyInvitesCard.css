.invite-card-container {
  display: grid;
  grid-template-columns: 40% 30% 30%;
  height: 100px;
  width: 100%;
  border: solid 3px black;
  border-radius: 10px;
  margin-bottom: 15px;
}

.invite-card-groupName {
  grid-column: 1/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
}

.invite-card-accept {
  grid-column: 2/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
}

.invite-card-reject {
  grid-column: 3/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
}