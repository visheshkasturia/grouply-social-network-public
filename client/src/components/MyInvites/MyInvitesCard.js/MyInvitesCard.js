import React from 'react';
import { acceptGroupInvite } from '../../../apiCalls/acceptGroupInvite';
import { rejectGroupInvite } from '../../../apiCalls/rejectGroupInvite';
import './MyInvitesCard.css';
/*eslint-disable */
function MyInvitesCard({ user, invite, getChangedUser }) {
  const handleAccept = async () => {
    acceptGroupInvite(invite, user.userName)
      .then((data) => {
        getChangedUser(data.invitations);
      })
      .catch((err) => {
        console.log(err)
      });
  }
  const handleReject = async () => {
    rejectGroupInvite(invite, user.userName)
      .then((data) => {
        getChangedUser(data.invitations);
      })
      .catch((err) => {
        console.log(err)
      });
  }

  return (
    <div className="invite-card-container">
      <div className="invite-card-groupName">
        {invite}
      </div>
      <div className="invite-card-accept">
        <button type="button" onClick={handleAccept} > Accept </button>
      </div>
      <div className="invite-card-reject">
        <button type="button" onClick={handleReject} > Reject </button>
      </div>
    </div>
  )
}

export default MyInvitesCard;
