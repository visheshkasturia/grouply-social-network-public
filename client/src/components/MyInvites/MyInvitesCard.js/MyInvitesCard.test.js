import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import MyInvitesCard from './MyInvitesCard'

const admin = 'abc'
const member = 'test1'
const post = '1'
const tag = 'tag'
const group = {
  groupName: "test",
  description: "desc",
  admins: [admin],
  allMembers: [member],
  allPosts: [post],
  tags: [tag]
}
const invite = 'abc'

const user = {
  _id: '1',
  invitations: ['abc'],
}

it('MyInvites renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MyInvitesCard invite={invite} user={user} />, div);
});

it('Members matches snapshot', () => {
  const tree = renderer.create(<MyInvitesCard invites={invite} user={user} />).toJSON();
  expect(tree).toMatchSnapshot();
});
