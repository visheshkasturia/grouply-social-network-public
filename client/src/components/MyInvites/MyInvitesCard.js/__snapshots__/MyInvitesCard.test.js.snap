// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members matches snapshot 1`] = `
<div
  className="invite-card-container"
>
  <div
    className="invite-card-groupName"
  />
  <div
    className="invite-card-accept"
  >
    <button
      onClick={[Function]}
      type="button"
    >
       Accept 
    </button>
  </div>
  <div
    className="invite-card-reject"
  >
    <button
      onClick={[Function]}
      type="button"
    >
       Reject 
    </button>
  </div>
</div>
`;
